File: src/main/kotlin/com/marcel/baumgardt/model/mapper/ShowingMapper.java
package com.marcel.baumgardt.model.mapper;

import com.marcel.baumgardt.model.db.Showing;
import com.marcel.baumgardt.model.dto.ShowingDate;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.time.OffsetTime;
import java.util.List;
import java.util.stream.Collectors;

@Component
public class ShowingMapper {

    public List<Showing> mapToShowings(Long cinemaId, Long movieId, List<ShowingDate> dates) {
        return dates.stream()
                .map(d -> getShowings(cinemaId, movieId, d))
                .flatMap(List::stream)
                .collect(Collectors.toList());
    }

    @NotNull
    private List<Showing> getShowings(Long cinemaId, Long movieId, ShowingDate date) {
        return date.getTimeList().stream()
                .map(of -> createShowing(cinemaId, movieId, date.getDay(), of))
                .collect(Collectors.toList());
    }

    private Showing createShowing(Long cinemaId, Long movieId, Integer day, OffsetTime offsetTime) {
        return new Showing();
    }
}

File: src/main/kotlin/com/marcel/baumgardt/model/mapper/ShowingMapper.kt
package com.marcel.baumgardt.model.mapper

import com.marcel.baumgardt.model.dao.ShowingDao
import com.marcel.baumgardt.model.db.Showing
import org.springframework.stereotype.Component
import java.util.stream.Collectors

@Component
class ShowingMapper {

    fun mapToShowings(showingsDao: List<ShowingDao>): List<Showing> {
        return showingsDao.stream()
            .map { dao -> mapToShowing(dao) }
            .collect(Collectors.toUnmodifiableList())
    }

    private fun mapToShowing(showingDao: ShowingDao): Showing {
        return Showing(
            showingDao.cinemaId,
            showingDao.movieId,
            showingDao.day,
            showingDao.offsetTime
        )
    }
}